### ISSUE TITLE
Address usability/accessibility for  "Use my location"

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#21976]
@mmiddaugh commented on [Tue Mar 23 2021](https://github.com/department-of-veterans-affairs/va.gov-vfs-teams/issues/55)

@mmiddaugh commented on [Tue Mar 09 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247)

## Issue Description
As a keyboard user, I want to be able to use the space bar to activate "Use my location" (rather than just the enter key) so that it is consistent with the usual keyboard navigation. 
As a Veteran using voice assistive technology, I want to be able to ask my tech to "Click on the use my location button" instead of "Click on the use my location link." so that I can access it better.  

"Use my location" should be written as a button <button> instead of a link <a> since the user is not navigating to a new page

---
## Tasks
- [x] Write "Use my location" as a button. 
- [x] Fix visual behavior of clear input and use my location button

## Acceptance Criteria (this includes feedback given in #17319)
- [x] Veterans navigating the Facility Locator using a keyboard can activate "Use my location" with the space bar. 
- [x] Veterans using voice assistive technology ask the tech to "Click on the Use my location button" instead of "Click on the Use my location link."
- [x] Clear input and use my location buttons do not shift to the right
- [x] The screen reader reads "finding your location" text while geolocation is in progress. 
- [x] "Use my location" should receive focus and be announced before "City state or postal code".
- [x] When geolocation completes, the focus goes to the City, state, postal code field (whether or not approval was given in the browser modal) and screenreader announces the field
- [x] If the Veteran moves into another field during the geolocation search process, that field receives focus (it is not "stolen" to focus on the city, state field)
- [x] If geolocation fails for any reason [an error message appears in a modal](https://design.va.gov/storybook/?path=/docs/components-modal--default): "Something went wrong. Please enter your location."
- [x] When the modal appears, the tab order is locked, the error message modal receives focus and is announced. 
- [x] Modal is dismissable by clicking the x and focus returns to the "City state postal field"
- [x] Data layer in place to track "clicks" on the feature 
- [x] Data layer to track the number of permission-related errors vs other types (needs analytics input)

https://dsva.slack.com/archives/C0FQSS30V/p1615814186014400?thread_ts=1615591295.004600&cid=C0FQSS30V

https://recordit.co/BmJjt9fWsS

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

---

@tlei123 commented on [Fri Mar 12 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247#issuecomment-797778820)

hi @lildziner , would you expect to have updated designs for this button?

if yes, will you be updating these two places in Zeplin?
- [Mobile](https://app.zeplin.io/project/600877fdb269d8a4f1f44efd/screen/600879364afaa6a6a2a8a827)
- [Desktop](https://app.zeplin.io/project/600877fdb269d8a4f1f44efd/screen/6008792b280d84a85bc6c015)

---

@mmiddaugh commented on [Tue Mar 23 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247#issuecomment-805246161)

We're not updating the design @tlei123 - just the coding

**Comment by mmiddaugh** [comment]
We will leave the country for now due to Veterans in Guam and outside the US.

**Comment by mpelzsherman** [comment]
@mmiddaugh Can you please provide clarification of the following AC items:

- Veterans can enable use of current location through hyperlink at search level
- When a Veteran clicks the link and enables the browser, the parameters populate the search field
- A "searching" indicator displays while the location is being identified (How is this different from what happens now?)

I would also like to see a design mockup for the error message modal, and I'd like @joshkimux to review and approve the design before we move forward.

**Comment by mmiddaugh** [comment]
We're anxious to get this feature across the finish line. 
Storybook has [this modal ](https://design.va.gov/storybook/?path=/docs/components-modal--default) which we think will work. @joshkimux will you please review our AC and the modal component when you have a moment?

**Comment by joshkimux** [comment]
@mmiddaugh From an accessibility standpoint this can be made compliant 👍 
The content piece may need more work to make error messaging more clear and actionable.

**Comment by mmiddaugh** [comment]
This feature is now in Staging. We'd love to get eyes on it from @joshkimux and @tlei123 when time allows. Thank you!

**Comment by joshkimux** [comment]
Wonderful work team @mpelzsherman @mmiddaugh  👏 👏 👏 

https://user-images.githubusercontent.com/14154792/117871807-d5043680-b26b-11eb-8bbe-daa65dc8895a.mov

One minor issue is that the modal's heading isn't announced if the user selects "don't allow" for some reason... but it works exactly as expected otherwise when opened again. I don't think this is a major or even moderate issue, so I think we can close out the ticket if no other bugs are caught by @tlei123  🥳 

https://user-images.githubusercontent.com/14154792/117873007-51e3e000-b26d-11eb-90a8-829475d22a3a.mov

**Comment by lildziner** [comment]
Question: what's the likelihood that the Medallia feedback button covers the 'close' button? On mobile I think the modal is fixed to the top of the page so that shouldn't be a problem but in this video it looks like both could be vertically centered at some breakpoints.

**Comment by mmiddaugh** [comment]
Thanks @lildziner and @joshkimux - I've asked Jason Woodman to take a look at your comments above - he was my point of contact for Medallia implementation. 

I'd like to keep the Medallia feedback button on this page - it's very important for assessing Veteran satisfaction with the Facility Locator - particularly since "completion" and "success" are so difficult to measure.

**Comment by mmiddaugh** [comment]
From Jason: 

- We did run Medallia through extensive 508 testing/fixes.
- We can exclude the feedback button on specific pages if that's the route deemed best.
- Moving the feedback button is not an option right now, but we could potentially look at floating it on the bottom of the screen (not sure this is possible or how that would work on mobile).
